### PR TITLE
12.0 afc 1541- final report

### PR DESCRIPTION
**Pull request by SebastianGieser-Vertel**

### CODE REVIEW

**Comment by SebastianGieser-Vertel** [comment]
Note that this version is not completely finished, but neither is the API (at least in my opinion). Might as well start testing in the meantime and get some feedback on the UX which (mostly) had no specification prior to implementation.

File: outplacement_final_report/models/outplacement.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    main_goal_id = fields.Many2one(comodel_name="outplacement.goal")
+    alternative_goal_id = fields.Many2one(comodel_name="outplacement.goal")
+
+class OutplacementGoal(models.Model):
+    _name = "outplacement.goal"
+
+    field_of_work_id = fields.Many2one(comodel_name="res.ssyk", string="Field of work") #return codes for these two
+    job_id = fields.Many2one(comodel_name="res.ssyk", string="Job title") #selection should be limited to search ilike field_of_work.code
+    job_description = fields.Char(string="Job description")
+    motivation = fields.Selection(selection=[('matches interest','Matchar deltagarens intressen'),('matches ability','Arbetsuppgifter matchar förmåga'),('market demand','Efterfrågan på arbetsmarknaden'),('complementing education','Kompletterar nuvarande utbildning'),('complementing experience','kompletterar tidigare erfarenheter'),('other', 'Other')], string="Motivation")
```
Comment (arturchilangwa-vertel) [inline outplacement_final_report/models/outplacement.py]: Long row, a bit hard to follow.

File: outplacement_final_report/models/outplacement.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    step_ids = fields.Many2many(comodel_name="outplacement.goal.step", string="Step")
+
+class OutplacementGoalStep(models.Model):
+    _name = "outplacement.goal.step"
+
+    #step type should be a series of boolean fields, but due to current API behaviour will be a selection field
+    step_type = fields.Selection(selection=[('study', 'Studera'),('regular education','Reguljär utbildning'),('fitting complementing efforts','Lämpliga kompletterande insatser'),('other','Other')], string="Type") #needs values
+    complementing_effort_type = fields.Selection(string="Complementing effort type", selection=[('study motivating effort','Studiemotiverande insats'),('prepare for work','Rusta inför arbete'),('match to work','Matcha till arbete'),('evaluate ability to work','Utreda arbetsförmågan'),('partake in education/internship/validation','Delta i en arbetsmarknadsutbildning/Praktik/Validering'),('Swedish studies in chosen field','Svenskastudier inom valt område'),('translation of grades','Översättning av betyg'),('evaluation and complementation of foreign education','Bedömning och komplettering av utländsk utbildning'),('other','Other')]) #might be better as a selection field
```
Comment (arturchilangwa-vertel) [inline outplacement_final_report/models/outplacement.py]: Long row, a bit hard to follow.

**Review by arturchilangwa-vertel — CHANGES_REQUESTED** [review]

**Review by anderswallenquist — COMMENTED** [review]
It would be better to divorce. Move api-code to module in odoo-api and leave code in glue-module

**Comment by arturchilangwa-vertel** [comment]
@SebastianGieser-Vertel 
tried to install `outplacement_final_report ` on machine but it did not work, here is the traceback:
```
2021-01-08 09:05:05,818 13909 WARNING outplacement_test odoo.addons.base.models.ir_model: Two fields (partner_state_id, stage_id) of outplacement() have the same label: State.
2021-01-08 09:05:05,907 13909 INFO outplacement_test odoo.modules.loading: loading outplacement_final_report/views/outplacement_view.xml
2021-01-08 09:05:05,971 13909 INFO outplacement_test odoo.addons.base.models.ir_ui_view: Field `level` does not exist

Error context:
View `outplacement.goal.step.form`
[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]
2021-01-08 09:05:05,974 13909 WARNING outplacement_test odoo.modules.loading: Transient module states were reset
2021-01-08 09:05:05,976 13909 ERROR outplacement_test odoo.modules.registry: Failed to load registry
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1128, in _validate_fields
    check(self)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 351, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 1138, in postprocess_and_fields
    self.raise_view_error(message, view_id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 568, in raise_view_error
    raise ValueError(message)
ValueError: Field `level` does not exist

Error context:
View `outplacement.goal.step.form`
[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nField `level` does not exist\n\nError context:\nView `outplacement.goal.step.form`\n[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 86, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 421, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 313, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package, report=report)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 68, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 802, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 865, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 764, in parse
    exc_info[2]
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Field `level` does not exist

Error context:
View `outplacement.goal.step.form`
[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]
None" while parsing /usr/share/odoo-outplacement/outplacement_final_report/views/outplacement_view.xml:55, near
<record model="ir.ui.view" id="view_outplacement_goal_step_form">
        <field name="name">outplacement.goal.step.form</field>
        <field name="model">outplacement.goal.step</field>
        <field name="arch" type="xml">
            <form>
                <group>
                    <field name="name"/>
                    <field name="step_type"/>
                    <field name="complementing_effort_type"/>
                    <field name="complementing_effort_description"/>
                    <field name="level"/>
                    <field name="start_date"/>
                    <field name="end_date"/>
                </group>
            </form>
        </field>
    </record>
2021-01-08 09:05:05,979 13909 CRITICAL outplacement_test odoo.service.server: Failed to initialize database `outplacement_test`.
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1128, in _validate_fields
    check(self)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 351, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 1138, in postprocess_and_fields
    self.raise_view_error(message, view_id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 568, in raise_view_error
    raise ValueError(message)
ValueError: Field `level` does not exist

Error context:
View `outplacement.goal.step.form`
[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nField `level` does not exist\n\nError context:\nView `outplacement.goal.step.form`\n[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/service/server.py", line 1162, in preload_registries
    registry = Registry.new(dbname, update_module=update_module)
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 86, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 421, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 313, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package, report=report)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 68, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 802, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 865, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 764, in parse
    exc_info[2]
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Field `level` does not exist

Error context:
View `outplacement.goal.step.form`
[view_id: 400, xml_id: n/a, model: outplacement.goal.step, parent_id: n/a]
None" while parsing /usr/share/odoo-outplacement/outplacement_final_report/views/outplacement_view.xml:55, near
<record model="ir.ui.view" id="view_outplacement_goal_step_form">
        <field name="name">outplacement.goal.step.form</field>
        <field name="model">outplacement.goal.step</field>
        <field name="arch" type="xml">
            <form>
                <group>
                    <field name="name"/>
                    <field name="step_type"/>
                    <field name="complementing_effort_type"/>
                    <field name="complementing_effort_description"/>
                    <field name="level"/>
                    <field name="start_date"/>
                    <field name="end_date"/>
                </group>
            </form>
        </field>
    </record>
2021-01-08 09:05:05,986 13909 INFO outplacement_test odoo.modules.loading: loading 1 modules...

```

**Comment by SebastianGieser-Vertel** [comment]
> It would be better to divorce. Move api-code to module in odoo-api and leave code in glue-module

Please elaborate on the code to be left in glue-module.

**Comment by arturchilangwa-vertel** [comment]
> > It would be better to divorce. Move api-code to module in odoo-api and leave code in glue-module
> 
> Please elaborate on the code to be left in glue-module.

Maybe we can make a issue-ticket for this? And merge this PR in the meantime.

**Comment by arturchilangwa-vertel** [comment]
Tried to install `outplacement_final_report`, but got this error while doing it. 
Seems that  `effort_type` is not a field or something. Or that `outplacement.goal.step.completing_effort`

```
Error:
Odoo Server Error

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1128, in _validate_fields
    check(self)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 351, in _check_xml
    self.postprocess_and_fields(view.model, view_doc, view.id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 1105, in postprocess_and_fields
    self.raise_view_error(_('Model not found: %(model)s') % dict(model=model), view_id)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 568, in raise_view_error
    raise ValueError(message)
ValueError: Model not found: outplacement.goal.step.completing_effort

Error context:
View `outplacement.goal.step.completing_effort.form`
[view_id: 396, xml_id: n/a, model: outplacement.goal.step.completing_effort, parent_id: n/a]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.exceptions.ValidationError: ('Error while validating constraint\n\nModel not found: outplacement.goal.step.completing_effort\n\nError context:\nView `outplacement.goal.step.completing_effort.form`\n[view_id: 396, xml_id: n/a, model: outplacement.goal.step.completing_effort, parent_id: n/a]', None)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 656, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 314, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 698, in dispatch
    result = self._call_function(**self.params)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 346, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/service/model.py", line 98, in wrapper
    return f(dbname, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 339, in checked_call
    result = self.endpoint(*a, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 941, in __call__
    return self.method(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/http.py", line 519, in response_wrap
    response = f(*args, **kw)
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 966, in call_button
    action = self._call_kw(model, method, args, {})
  File "/usr/lib/python3/dist-packages/odoo/addons/web/controllers/main.py", line 954, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 759, in call_kw
    return _call_kw_multi(method, model, args, kwargs)
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 746, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "<decorator-gen-61>", line 2, in button_immediate_install
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 74, in check_and_log
    return method(self, *args, **kwargs)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 445, in button_immediate_install
    return self._button_immediate_function(type(self).button_install)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_module.py", line 561, in _button_immediate_function
    modules.registry.Registry.new(self._cr.dbname, update_module=True)
  File "/usr/lib/python3/dist-packages/odoo/modules/registry.py", line 86, in new
    odoo.modules.load_modules(registry._db, force_demo, status, update_module)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 421, in load_modules
    loaded_modules, update_module, models_to_check)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 313, in load_marked_modules
    perform_checks=perform_checks, models_to_check=models_to_check
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 222, in load_module_graph
    load_data(cr, idref, mode, kind='data', package=package, report=report)
  File "/usr/lib/python3/dist-packages/odoo/modules/loading.py", line 68, in load_data
    tools.convert_file(cr, package.name, filename, idref, mode, noupdate, kind, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 802, in convert_file
    convert_xml_import(cr, module, fp, idref, mode, noupdate, report)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 865, in convert_xml_import
    obj.parse(doc.getroot(), mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 764, in parse
    exc_info[2]
  File "/usr/lib/python3/dist-packages/odoo/tools/pycompat.py", line 86, in reraise
    raise value.with_traceback(tb)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 758, in parse
    self._tags[rec.tag](rec, de, mode=mode)
  File "/usr/lib/python3/dist-packages/odoo/tools/convert.py", line 663, in _tag_record
    record = model.with_context(rec_context)._load_records([data], self.mode == 'update')
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3908, in _load_records
    records = self._load_records_create([data['values'] for data in to_create])
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3822, in _load_records_create
    return self.create(values)
  File "<decorator-gen-32>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 428, in create
    return super(View, self).create(vals_list)
  File "<decorator-gen-3>", line 2, in create
  File "/usr/lib/python3/dist-packages/odoo/api.py", line 462, in _model_create_multi
    return create(self, arg)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3639, in create
    fields[0].determine_inverse(batch_recs)
  File "/usr/lib/python3/dist-packages/odoo/fields.py", line 1168, in determine_inverse
    getattr(records, self.inverse)()
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 275, in _inverse_arch
    view.write(data)
  File "/usr/lib/python3/dist-packages/odoo/addons/base/models/ir_ui_view.py", line 445, in write
    res = super(View, self).write(self._compute_defaults(vals))
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3364, in write
    self._write(store_vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 3500, in _write
    self._validate_fields(vals)
  File "/usr/lib/python3/dist-packages/odoo/models.py", line 1132, in _validate_fields
    raise ValidationError("%s\n\n%s" % (_("Error while validating constraint"), tools.ustr(e)))
odoo.tools.convert.ParseError: "Error while validating constraint

Model not found: outplacement.goal.step.completing_effort

Error context:
View `outplacement.goal.step.completing_effort.form`
[view_id: 396, xml_id: n/a, model: outplacement.goal.step.completing_effort, parent_id: n/a]
None" while parsing /usr/share/odoo-outplacement/outplacement_final_report/views/outplacement_view.xml:73, near
<record model="ir.ui.view" id="view_outplacement_goal_step_completing_effort_form">
        <field name="name">outplacement.goal.step.completing_effort.form</field>
        <field name="model">outplacement.goal.step.completing_effort</field>
        <field name="arch" type="xml">
            <form>
                <group>
                    <field name="effort_type"/>
                </group>
            </form>
        </field>
    </record>
```

**Review by arturchilangwa-vertel — APPROVED** [review]
